src/js/netjsongraph.render.js

```diff
@@ -380,12 +380,17 @@ class NetJSONGraphRender {
     if (self.type === "geojson") {
       self.leaflet.geoJSON = L.geoJSON(self.data, self.config.geoOptions);
 
-      if (self.config.clustering) {
+      // Check if clustering should be applied based on current zoom level and configuration
+      const shouldApplyClustering =
```

Comment:
I think we can make this more concise and clearer:
```suggestion
      const needsClustering =
```